nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -392,24 +418,33 @@ def _test_binary(op, arrseq, scalars, seqs, inplace=False):
                 continue  # Going to deal with it separately.
 
             _test_binary(op, seq_int, [42, -3, True, 0], [seq_int, seq_bool, -seq_int], inplace=True)  # int <-- int
-            assert_raises(TypeError, _test_binary, op, seq_int, [0.5], [], inplace=True)  # int <-- float
-            assert_raises(TypeError, _test_binary, op, seq_int, [], [seq], inplace=True)  # int <-- float
+            
+            with pytest.raises(TypeError):
+                _test_binary(op, seq_int, [0.5], [], inplace=True) # int <-- float
+                _test_binary(op, seq_int, [], [seq], inplace=True) # int <-- float
```

Comment:
`pytest.raises()` will exit the context on the first type error. We should use it for each erroring call.

```suggestion
           with pytest.raises(TypeError):
                _test_binary(op, seq_int, [], [seq], inplace=True) # int <-- float
```